applications/StructuralMechanicsApplication/custom_constitutive/linear_elastic_orthotropic_2D_law.cpp

```diff
@@ -0,0 +1,422 @@
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Peter Wilson
+//       Contact:    A.Winterstein@tum.de
+//
+
+// System includes
+#include <iostream>
+
+// External includes
+#include<cmath>
+
+// Project includes
+#include "includes/properties.h"
+#include "custom_constitutive/linear_elastic_orthotropic_2D_law.hpp"
+
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+	//******************************CONSTRUCTOR*********************************
+	//**************************************************************************
+
+	LinearElasticOrthotropic2DLaw::LinearElasticOrthotropic2DLaw()
+		: ConstitutiveLaw()
+	{
+	}
+
+	//******************************COPY CONSTRUCTOR****************************
+	//**************************************************************************
+
+	LinearElasticOrthotropic2DLaw::LinearElasticOrthotropic2DLaw
+		(const LinearElasticOrthotropic2DLaw& rOther)
+		: ConstitutiveLaw(rOther)
+		, mInverseDeformationGradientF0(rOther.mInverseDeformationGradientF0)
+		, mDeterminantF0(rOther.mDeterminantF0)
+		, mStrainEnergy(rOther.mStrainEnergy)
+	{
+	}
+
+	//********************************CLONE*************************************
+	//**************************************************************************
+
+	ConstitutiveLaw::Pointer LinearElasticOrthotropic2DLaw::Clone() const
+	{
+		LinearElasticOrthotropic2DLaw::Pointer p_clone
+			(new LinearElasticOrthotropic2DLaw(*this));
+		return p_clone;
+	}
+
+	//*******************************DESTRUCTOR*********************************
+	//**************************************************************************
+
+	LinearElasticOrthotropic2DLaw::~LinearElasticOrthotropic2DLaw()
+	{
+	}
+
+	void LinearElasticOrthotropic2DLaw::testString()
+	{
+		std::cout << "Printing LinearElasticOrthotropic2DLaw test string" << std::endl;
+	}
+
+	//*******************************OPERATIONS FROM BASE CLASS***************************
+	//************************************************************************************
+
+	//***********************HAS : DOUBLE - VECTOR - MATRIX*******************************
+	//************************************************************************************
+	
+	// pwdebug
+	bool LinearElasticOrthotropic2DLaw::Has(const Variable<double>& rThisVariable)
+	{
+		return false;
+	}
+
+	bool LinearElasticOrthotropic2DLaw::Has(const Variable<Vector>& rThisVariable)
+	{
+		return false;
+	}
+
+	bool LinearElasticOrthotropic2DLaw::Has(const Variable<Matrix>& rThisVariable)
+	{
+		return false;
+	}
+	
+
+	//***********************GET VALUE: DOUBLE - VECTOR - MATRIX**************************
+	//************************************************************************************
+	
+	// pwdebug
+	double& LinearElasticOrthotropic2DLaw::GetValue(const Variable<double>& rThisVariable, double& rValue)
+	{
+		if (rThisVariable == STRAIN_ENERGY)
```

Comment:
there is a function in the cl which allows u to calculate the strain_energy instead of having to store it

see

https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/constitutive_law.h#L652